pkg/kubelet/pluginmanager/plugin_manager_test.go

```diff
@@ -134,19 +137,28 @@ func TestPluginRegistration(t *testing.T) {
 	fakeHandler := newFakePluginHandler()
 	pluginManager.AddHandler(registerapi.DevicePlugin, fakeHandler)
 
-	// Add a new plugin
-	socketPath := fmt.Sprintf("%s/plugin.sock", socketDir)
-	pluginName := "example-plugin"
-	p := pluginwatcher.NewTestExamplePlugin(pluginName, registerapi.DevicePlugin, socketPath, supportedVersions...)
-	require.NoError(t, p.Serve("v1beta1", "v1beta2"))
-
-	// Verify that the plugin is registered
-	waitForRegistration(t, fakeHandler)
+	const (
+		pluginName = "example-plugin"
+		maxDepth   = 3
+	)
+	// Make sure the plugin manager is aware of the socket in subdirectories
+	for i := 0; i < maxDepth; i++ {
+		pluginDir := socketDir
+		for j := 0; j < i; j++ {
+			pluginDir = filepath.Join(pluginDir, strconv.Itoa(j))
+		}
+		_ = os.MkdirAll(pluginDir, os.ModePerm)
+		socketPath := filepath.Join(pluginDir, "plugin.sock")
+		// Add a new plugin
+		p := pluginwatcher.NewTestExamplePlugin(pluginName, registerapi.DevicePlugin, socketPath, supportedVersions...)
+		require.NoError(t, p.Serve("v1beta1", "v1beta2"))
+
+		// Verify that the plugin is registered
+		waitForRegistration(t, fakeHandler)
```

Comment:
waitForRegistration only verifies a single plugin was registered, so after the first one at depth 0, it is not checking anything further... we need to pass in the particular path we're expecting to be registered, right?

Reply:
we don't need to pass the particular path, we just need to reset the handler.

Reply:
> we don't need to pass the particular path, we just need to reset the handler.

it would be better to check the specific path, otherwise a late/duplicate event from an early plugin could make it look like a later one passed

Reply:
Actually a late event would not affect the test, as the test would fail immediately if registration timeout, and I think duplicate event rarely occurs on a local network. But I think this check is good to have.